pkg/controller/mizar/mizar-arktos-network-controller.go

```diff
@@ -172,221 +168,192 @@ func (c *MizarArktosNetworkController) syncNetwork(eventKeyWithType KeyWithEvent
 		return err
 	}
-	klog.V(4).Infof("Get Name : (%s) and GVK: (%s)", unstructuredObj.GetName(), gvk.String())
 
 	// Get mapping from GVK for GVR (Group Version Resource) used by dynamic client resource
 	mapper := restmapper.NewDeferredDiscoveryRESTMapper(memory.NewMemCacheClient(discoveryClient))
-	klog.V(4).Infof("Name: %s - GVK group kind : (%v) - GVK version: (%v)", unstructuredObj.GetName(), gvk.GroupKind(), gvk.Version)
 	mapping, err := mapper.RESTMapping(gvk.GroupKind(), gvk.Version)
 
-	klog.V(4).Infof("Name: %s - GVK group kind : (%s) - GVK version: (%s)", unstructuredObj.GetName(), unstructuredObj.GetKind(), unstructuredObj.GetAPIVersion())
 	if err != nil {
 		klog.Errorf("Get mapping between GVK and GVR in error (%v).", err)
 		return err
 	}
 
-	klog.V(4).Infof("Name: %s - get mapping scope name: (%s) - meta RESTScopeNameNamespace: (%s)", unstructuredObj.GetName(), mapping.Scope.Name(), meta.RESTScopeNameNamespace)
-
 	// Create dynamic client resource
 	var dynamicClientResource dynamic.ResourceInterface
 	if mapping.Scope.Name() == meta.RESTScopeNameNamespace {
 		// namespaced resources should specify the namespace under tenant
+		// VPC/Subnet is tenant unaware
 		if unstructuredObj.GetNamespace() == "" {
-			unstructuredObj.SetNamespace("default")
+			unstructuredObj.SetNamespace(metav1.NamespaceDefault)
 		}
 		namespace := unstructuredObj.GetNamespace()
-		klog.V(4).Infof("Mapping resource: (%v) - set tenant: (%s) - namespace : (%s)", mapping.Resource, "system", namespace)
-		dynamicClientResource = dynamicClient.Resource(mapping.Resource).NamespaceWithMultiTenancy(namespace, "system")
+		dynamicClientResource = dynamicClient.Resource(mapping.Resource).NamespaceWithMultiTenancy(namespace, metav1.TenantSystem)
 
 	} else {
 		// for cluster-wide resources
 		dynamicClientResource = dynamicClient.Resource(mapping.Resource)
 	}
 
 	// Create CRD resource - vpc or subnet
-	actualObject, err := dynamicClientResource.Create(unstructuredObj, metav1.CreateOptions{})
+	_, err = dynamicClientResource.Create(unstructuredObj, metav1.CreateOptions{})
 
-	if err == nil {
-		klog.V(4).Infof("Get actual object's name : (%s)", actualObject.GetName())
-		klog.V(4).Infof("Get actual object's GVK : (%v)", actualObject.GroupVersionKind())
-		klog.V(4).Infof("Get actual object's objectKind : (%v)", actualObject.GetObjectKind())
-	} else {
-		klog.Errorf("Create actual object's name: (%s) in error (%v).", unstructuredObj.GetName(), err)
+	if err != nil {
+		klog.Errorf("Create unstructed object (%s) result in error (%v).", unstructuredObj.GetName(), err)
 	}
 
 	return err
 }
 
-func GetCRDVpcOrSubnetSpec(defaultTemplatePath, vpcOrSubnetName, tenant string) ([]byte, error) {
-	// For updating the data in default vpc/subnet template
-	var availableData = map[string]string{
-		"Tenant": tenant,
+func generateVPCSpec(vpcName string) (int, *MizarVPC) {
+	// randomize ip start segment:
+	ipStart := ran.Intn(255) + 1 	// IpStart range [1, 255]
```

Comment:
what if user specified IP range for the VPC?

Reply:
> what if user specified IP range for the VPC?

We have agreed the scope for 130, there will be limited support for VPC. One VPC per tenant. Also, not supporting multiple subnets for a single VPC. User specified IP range is out of 130 scope.